tools/pre-commit

```diff
@@ -9,17 +9,17 @@
 # contains your provisioned Zulip development environment, the linter
 # will automatically be run through `vagrant ssh`.
 
-changed_files=$(git diff --cached --name-only --diff-filter=ACM)
-if [ -z "$changed_files" ]; then
+mapfile -t -d '' changed_files < <(git diff -z --cached --name-only --diff-filter=ACM)
```

Comment:
For me, this throws this:
```
mapfile: usage: mapfile [-n count] [-O origin] [-s count] [-t] [-u fd] [-C callback] [-c quantum] [array]
```

(Use `tools/setup-git-repo` and then make a commit to test this one)

Reply:
Blargh, `mapfile -d` is new in Bash 4.4—maybe binary-safety is too much to ask for.

Reply:
Fixed.